### PR TITLE
Implemented all the required methods in the MyLinkedList class

### PR DESCRIPTION
**Pull request by dima-semeniuk**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -3,45 +3,142 @@
...
+            Node<T> newNode = new Node<>(value, null, tail);
+            tail.next = newNode;
+            tail = tail.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
```
Comment (yurii-tomashevskyi) [inline src/main/java/core/basesyntax/MyLinkedList.java:44]: beter create method checkElementIndex(int index). You use this if statement - 4 times

**Review by yurii-tomashevskyi — APPROVED** [review]
Good job 🏆🔥

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -3,45 +3,137 @@
...
+        } else if (index == size - 1) {
+            tail.previous.next = tail.next;
+            tail = tail.previous;
+        } else {
+            Node<T> currentNode = getNodeByIndex(index);
+            currentNode.previous.next = currentNode.next;
+            currentNode.next.previous = currentNode.previous;
+        }
+        size--;
+        return removedNodeValue;
```
Comment (AlexRozv) [inline src/main/java/core/basesyntax/MyLinkedList.java:113]: https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#create-a-separate-method-unlinknode-node

let's create a separate method that will unlink our node

also, we can simplify this like this:
we can create temporary nodes node.prev and node.next, then do two checks:
if (prev == null) {...}
else {...}
if (next == null) {...}
else {...}

**Review by AlexRozv — CHANGES_REQUESTED** [review]

**Review by AlexRozv — APPROVED** [review]
good job 👍

**Pull request closed without merging**